src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -361,7 +361,8 @@ public class GenerateProtoTask extends DefaultTask {
     Preconditions.checkState(state == State.FINALIZED, 'doneConfig() has not been called')
 
     ToolsLocator tools = project.protobuf.tools
-    Set<File> protoFiles = inputs.sourceFiles.files
+    // Sort to make sure files are in a consistent order
+    List<File> protoFiles = inputs.sourceFiles.files.sort()
```

Comment:
I still don't understand how the order the files are written would impact up-to-date checks. Can you elaborate?